lib/appsignal/phoenix/instrumenter.ex

```diff
@@ -18,68 +18,49 @@ if Appsignal.phoenix? do
-    You might be using your endpoint's `instrument/4` macro to create
-    custom instrumentation. If you want those events to become part of
-    the AppSignal timeline as well, you need to create a custom
-    instrumenter module with the help of
-    Appsignal.Phoenix.InstrumenterDSL:
-
-    ```
-    defmodule PhoenixApp.CustomInstrumenter do
-
-      import Appsignal.Phoenix.InstrumenterDSL
-      instrumenter :phoenix_controller_call
-      instrumenter :phoenix_controller_render
-      instrumenter :custom_event
-      instrumenter :another_custom_event
-    end
-    ```
-
-    And then, use that instead of the AppSignal instrumenter in your `config.exs`:
-
-    ```
-    config :phoenix_app, PhoenixApp.Endpoint,
-      instrumenters: [PhoenixApp.CustomInstrumenter]
-    ```
-
     See the [Phoenix integration guide](phoenix.html) for information on
     how to instrument other aspects of Phoenix.
     """
+    @transaction Application.get_env(:appsignal, :appsignal_transaction, Appsignal.Transaction)
 
-    alias Appsignal.Transaction
+    @doc false
+    def phoenix_controller_call(:start, _, args), do: start_event(args)
 
-    require Logger
+    @doc false
+    def phoenix_controller_call(:stop, _diff, {%Appsignal.Transaction{} = transaction, %{conn: conn} = args}) do
+      finish_event(transaction, "phoenix_controller_call", args)
 
-    import Appsignal.Phoenix.InstrumenterDSL
-    instrumenter :phoenix_controller_call
-    instrumenter :phoenix_controller_render
+      @transaction.try_set_action(transaction, conn)
 
-    @doc false
-    def maybe_transaction_start_event(%Transaction{} = transaction, args) do
-      {Transaction.start_event(transaction), args}
-    end
-    def maybe_transaction_start_event(pid, args) when is_pid(pid) do
-      maybe_transaction_start_event(Appsignal.TransactionRegistry.lookup(pid), args)
-    end
-    def maybe_transaction_start_event(%{conn: %Plug.Conn{} = conn}, args) do
-      maybe_transaction_start_event(conn.assigns[:appsignal_transaction], args)
-    end
-    def maybe_transaction_start_event(%{}, args) do
-      maybe_transaction_start_event(self(), args)
+      if @transaction.finish(transaction) == :sample do
+        @transaction.set_request_metadata(transaction, conn)
+      end
+
+      :ok = @transaction.complete(transaction)
     end
-    def maybe_transaction_start_event(nil, _), do: nil
+    def phoenix_controller_call(:stop, _, _), do: nil
 
     @doc false
-    def maybe_transaction_finish_event(_event, nil), do: nil
-    def maybe_transaction_finish_event(event, {transaction, args}) do
-      Transaction.finish_event(transaction, event, event, args, 0)
-    end
+    def phoenix_controller_render(:start, _, args), do: start_event(args)
 
     @doc false
-    def cleanup_args(args) do
-      Map.delete(args, :conn)
+    def phoenix_controller_render(:stop, _diff, {%Appsignal.Transaction{} = transaction, args}) do
+      finish_event(transaction, "phoenix_controller_render", args)
```

Comment:
We discussed changing this name so that it doesn't get put in the "other" group. Can we change this to the event name `controller_render.phoenix`?